INT B-20437 Send Destination GBLOC to Prime

brooklyn-welsh opened the pull request:
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20437)

## Summary

This branch updates all Prime API versions to send the destination GBLOC of a move to the Prime, which will allow the prime to get contact info for the destination office. This branch is only for backend API changes, there is currently no space on the frontend for this information to be displayed.

### How to test

1. Access any move in the Prime Simulator role
2. Bring up dev tools and refresh if needed, go to the Network tab.
3.Click on the item with the move's UUID, under the "Order" object, you should now see a "destinationDutyLocationGBLOC" item with a GBLOC string value.


## Screenshots
![primedestgbloc](https://github.com/user-attachments/assets/9c10111e-0b1f-4381-8820-451917dd2b64)

taeJungCaci commented:
Some of the things I've noticed - Since you made the changes to store the destinationGBLOC in orders table, you may have to consider the case when the destinationDutyLocation is edited by the SC/TOO. Will it need to update the DestinationGBLOC in the table? Also, since this destinationGBLOC only gets stored when Order is created, you may have to manually sent one to Prime API if the entry is null using `FetchGBLOCForPostalCode` in the model_to_payload maybe.

taeJungCaci commented on pkg/handlers/primeapi/payloads/model_to_payload.go:
```diff
@@ -159,23 +159,25 @@ func Order(order *models.Order) *primemessages.Order {
 
 	if strings.ToLower(payload.Customer.Branch) == "marines" {
 		payload.OriginDutyLocationGBLOC = "USMC"
+		payload.DestinationDutyLocationGBLOC = "USMC"
```
Just double checking if this is true - DestinationDutyLocationGBLOC also USMC?

brooklyn-welsh replied:
That might actually be a question worth asking Gary, I assumed that since the member is still a part of the USMC branch that it would remain USMC regardless of destination? But there might be a case where that isn't true.

brooklyn-welsh replied:
Gary responded: the team concurs that there should be no instance where a move would change a Marines service member's GBLOC.

brooklyn-welsh commented:
> Some of the things I've noticed - Since you made the changes to store the destinationGBLOC in orders table, you may have to consider the case when the destinationDutyLocation is edited by the SC/TOO. Will it need to update the DestinationGBLOC in the table? Also, since this destinationGBLOC only gets stored when Order is created, you may have to manually sent one to Prime API if the entry(currently existing moves) is null using `FetchGBLOCForPostalCode` in the model_to_payload I think.

Good point, I think you're right, I'll need to make sure that any edits to the destination, anywhere in the app, also check and recalculate the GBLOC

brooklyn-welsh commented:
> > Some of the things I've noticed - Since you made the changes to store the destinationGBLOC in orders table, you may have to consider the case when the destinationDutyLocation is edited by the SC/TOO. Will it need to update the DestinationGBLOC in the table? Also, since this destinationGBLOC only gets stored when Order is created, you may have to manually sent one to Prime API if the entry(currently existing moves) is null using `FetchGBLOCForPostalCode` in the model_to_payload I think.
> 
> Good point, I think you're right, I'll need to make sure that any edits to the destination, anywhere in the app, also check and recalculate the GBLOC

Added code that should cover edits by office users and by service member, and added a code section in "FetchMoveTaskOrder" handler, which is used by the Prime API, that generates the GBLOC if it isn't present.

taeJungCaci reviewed (approved):
LGTM

WeatherfordAaron reviewed (approved):
Sorry it took me a bit on this one.. everything lgtm 👽
